packages/flutter_tools/lib/src/commands/daemon.dart

```diff
@@ -416,36 +416,129 @@ class DaemonDomain extends Domain {
       final Set<SupportedPlatform> supportedPlatforms = flutterProject.getSupportedPlatforms().toSet();
-      if (featureFlags.isLinuxEnabled && supportedPlatforms.contains(SupportedPlatform.linux)) {
-        result.add('linux');
-      }
-      if (featureFlags.isMacOSEnabled && supportedPlatforms.contains(SupportedPlatform.macos)) {
-        result.add('macos');
-      }
-      if (featureFlags.isWindowsEnabled && supportedPlatforms.contains(SupportedPlatform.windows)) {
-        result.add('windows');
-      }
-      if (featureFlags.isIOSEnabled && supportedPlatforms.contains(SupportedPlatform.ios)) {
-        result.add('ios');
-      }
-      if (featureFlags.isAndroidEnabled && supportedPlatforms.contains(SupportedPlatform.android)) {
-        result.add('android');
-      }
-      if (featureFlags.isWebEnabled && supportedPlatforms.contains(SupportedPlatform.web)) {
-        result.add('web');
-      }
-      if (featureFlags.isFuchsiaEnabled && supportedPlatforms.contains(SupportedPlatform.fuchsia)) {
-        result.add('fuchsia');
-      }
-      if (featureFlags.areCustomDevicesEnabled) {
-        result.add('custom');
+
+      void handlePlatformType(
+        PlatformType platform,
+      ) {
+        bool isSupported = true;
+        bool canBeEnabledWithCreate = true;
+        final List<String> reasons = <String>[];
+        switch (platform) {
+          case PlatformType.linux:
+            if (!featureFlags.isLinuxEnabled) {
+              isSupported = false;
+              reasons.add('the Linux feature is not enabled ("flutter config --enable-linux-desktop")');
+              canBeEnabledWithCreate = false;
+            }
+            if (!supportedPlatforms.contains(SupportedPlatform.linux)) {
+              isSupported = false;
+              reasons.add('the current Flutter project does not have a Linux platform directory');
+            }
+          case PlatformType.macos:
+            if (!featureFlags.isMacOSEnabled) {
+              isSupported = false;
+              reasons.add('the macOS feature is not enabled ("flutter config --enable-macos-desktop")');
+              canBeEnabledWithCreate = false;
+            }
+            if (!supportedPlatforms.contains(SupportedPlatform.macos)) {
+              isSupported = false;
+              reasons.add('the current Flutter project does not have a macOS platform directory');
+            }
+          case PlatformType.windows:
+            if (!featureFlags.isWindowsEnabled) {
+              isSupported = false;
+              reasons.add('the Windows feature is not enabled ("flutter config --enable-windows-desktop")');
+              canBeEnabledWithCreate = false;
+            }
+            if (!supportedPlatforms.contains(SupportedPlatform.windows)) {
+              isSupported = false;
+              reasons.add('the current Flutter project does not have a Windows platform directory');
+            }
```

Comment:
There's quite a lot of repetition here, maybe it could be simplified by enumerating some records with each of the values?

```dart
final x = [
  ("Windows", PlatformType.windows, featureFlags.isWindowsEnabled, SupportedPlatform.windows, "windows-desktop"),
  ...
];
```